### DISCUSSION TITLE
How can I apply a quickfix

### DISCUSSION THREAD

**Discussion by prabanjanraja** [LazyVim/LazyVim#1660] [Q&A]
I am using lazyvim for cpp projects
Please help me with these questions

How can I apply the quick fix from the trouble window?
Why is the Quickfix window always empty, even when I see (fix available) in trouble window?
What is the difference between code action and source action?

**Comment by 0x2E** [comment]
> How can I apply the quick fix from the trouble window?

Jump to the line of code with `(fix available)` at the end of its diagnostics, press `<Leader> ca` to show code actions, choose one.

> Why is the Quickfix window always empty, even when I see (fix available) in trouble window?

https://github.com/folke/trouble.nvim/issues/19#issuecomment-831500833 .

> What is the difference between code action and source action?

`Code Action` shows the lsp code actions available at the current cursor position. `Source Action` shows the lsp code actions available for entire file.

- https://github.com/LazyVim/LazyVim/blob/879e29504d43e9f178d967ecc34d482f902e5a91/lua/lazyvim/plugins/lsp/keymaps.lua#L25-L41
- https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#codeActionKind
